src/concurrent/channel.cr

```diff
@@ -269,4 +271,11 @@ class Channel::Unbuffered(T) < Channel(T)
   def full?
     @has_value || @receivers.empty?
   end
+
+  def close
+    super
+    if sender = @sender
+      Scheduler.enqueue sender
```

Comment:
I'm not sure but `@sender` may have been be already enqueued, see https://github.com/carlhoerberg/crystal/blob/c901b8096fff57893c46555401a395ca1c0a1aef/src/concurrent/channel.cr#L263. Maybe `@sender` should be set to `nil` in `#receive_impl`?

```crystal
@value.tap do
  @has_value = false
  Scheduler.enqueue @sender.not_nil!
  @sender = nil
end
```